CONTRIBUTING.md

```diff
@@ -0,0 +1,104 @@
+# Contributing Guidelines
+
+Welcome, and thanks for your interest in contributing to the `snowflake` repository! :snowflake:
+
+By contributing, you are agreeing that we may redistribute your work under this [license](https://github.com/Sage-Bionetworks/snowflake/tree/snow-90-auto-db-clone?tab=License-1-ov-file#).
+
+## Getting Started
+
+To start contributing, follow these steps to set up and develop on your local repository:
+
+### 1. Clone the Repository
+
+```bash
+git clone https://github.com/Sage-Bionetworks/snowflake
+```
+
+### 2. Fetch the Latest `dev` Branch
+
+After cloning, navigate to the repository directory:
+
+```bash
+cd snowflake
+```
+
+Then, fetch the latest updates from the `dev` branch to ensure you’re working with the latest codebase:
+
+```bash
+git fetch origin dev
+```
+
+### 3. Create a New Branch Off `dev`
+
+Create and checkout your feature branch from the latest `dev` branch. Name it based on the Jira ticket number and your feature/fix. For example:
+
+```bash
+git checkout -b snow-123-new-feature origin/dev
+```
+
+Your branch will now be tracking `origin/dev` which you can merge with or rebase onto should a merge conflict occur. For more guidance
+on how to resolve merge conflicts, [see here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/about-merge-conflicts#resolving-merge-conflicts).
+
+> [!IMPORTANT]
+> If you plan to run the automated testing described in section [Running CI Jobs for Database Testing](#running-ci-jobs-for-database-testing), your branch name needs to start with `snow-`,
+> otherwise the test deployment will fail.
+
+### 4. Push to The Remote Branch
+
+Once you've made your changes and committed them locally, push your branch to the remote repository:
+
+```
+git push origin snow-123-new-feature
+```
+
+### 5. Create a Draft Pull Request
+
+In order to initiate automated testing you will need to work on a draft pull request (PR) on GitHub. After pushing your commits to
+the remote branch in Step 4, use the GitHub UI to initate a PR and convert it to draft mode.
```

Comment:
This seems optional. If so, can we make that more explicit?

Reply:
I would say creating the draft PR isn't optional, since you would want the `schemachange` test to pass before marking your PR as ready for review. What I'll add is a separate section for "mark your PR as ready for review after tests pass".

Reply:
https://github.com/Sage-Bionetworks/snowflake/pull/78/commits/ab9297a712bf260fa53f7e5dc247fc1acd85a2c8

Reply:
Ah, okay makes sense. In RECOVER we've been doing this a bit differently. Tests run as part of a regular PR (no need to remember to convert to a draft) and if the tests fail, then there's an implicit understanding that a review isn't needed yet. Although, tests failing as part of a PR is pretty rare -- since we run tests whenever there is a new commit to a feature branch -- issues tend to get caught before a PR is created / a review is needed.

Reply:
> Tests run as part of a regular PR (no need to remember to convert to a draft)

Makes sense! Technically, the job can run on a non-draft PR, but it's good practice to mark PRs as drafts when they're not in their final state (all tests passing, etc), so I think we should enforce this in the workflow.